new-frontiers-14/frontier-station-14
# Smuggling posters are now random and NFSD gets rewarded for getting prints of posters and dead drops

## About the PR


The PR changes several things about the smuggling posters and their mechanics:

- Changed their activation rate down from a of 15-90 minutes range to a 5-10 minutes range and made the distance that the drop pod is spawned at smaller as well: 362ebec , 7670a6d, 3bcca71, d454abd

![image](https://github.com/user-attachments/assets/9b000ebb-56ec-4a75-857a-a2e24dac786a)

- Only 2 dead drop posters can be active at a time (have the verb to spawn the paper with the dead drop) as per the request of @whatston3: d454abd

> lots of good stuff in here. With the whole '1 active per POI' deal, i was actually considering expanding this to be more of a sector-wide phenomenon in order to both A. Slow down the rate of the dead drops being used and B. Have it be 'available' more frequently to the players who do participate, so that they dont ever feel like they are just 'waiting'.
> 
> So, instead of a per-POI basis, you could theoretically re-write it so that there are only ever X amount active _server-wide_. In that way, we can mostly forego any kind of timer limit (probably still keep a short one to prevent spam/hyper abuse), while still gauranteeing it is 'available' at any given time, while also still providing the challenge of the players still having to venture out and search multiple locations/posters for wherever the next 'drop' may have activated from.
> 
> Thoughts on making it more a sector-wide "2 active at a time" limit?

- NFSD is now warned about the location of dead drops after 15 minutes have passed: 7670a6d

![image](https://github.com/user-attachments/assets/901734e8-ef19-412f-ab4c-462c880db8b6)

- NFSD get partial hints about the location of each POI, 1st time no hint, 2nd time they get fed two locations, one being fake and 3rd time it's the actual location: 3bcca71

![image](https://github.com/user-attachments/assets/82ad353c-6c6b-4862-a7d8-27873e52828c)


- Made every poster on tinnia.yml, arena.yml and caseyscasino.yml roll for being a DeadDrop poster, this poster abundance is not an issue as only 1 can be active per POI: 7c80217 , 5f6b3f3

![image](https://github.com/user-attachments/assets/ace3d86f-4c05-4b39-be37-afb004eaaadc)

- NFSD detectives are now rewarded for scanning any used contraband posters with 3 FUC and any spawned dead drops with 6 FUC with a forensic scanner: 27dc7f6 , 470fd58, a47f678

![image](https://github.com/user-attachments/assets/e5f90085-348d-43d7-a245-19fd6b480008)
![image](https://github.com/user-attachments/assets/6c128b42-d331-466b-98b6-635000cad3a8)

- New guidebook paragraph in the Forensics entry detailing how to get prints of posters, Freelance now has a 1/3 chance of getting the smuggler's current ship ID after 15 minutes and lowered the chance of posters spawning back from 0.9 to 0.5: 3fee8c8

![image](https://github.com/user-attachments/assets/0ce64592-3b0e-4aec-abda-5e16d7bebb0e)
![image](https://github.com/user-attachments/assets/0f4ae56a-6778-41d9-a343-0c92c9e659e9)


## Why / Balance


Discord idea thread: https://discord.com/channels/1123826877245694004/1260704805282910321

Currently the dead drop system or "smuggling" is not worth doing for newer players, a lot of the profitability from them comes from the fact whether you know EXACTLY which posters to search on each of the main locations and whether you get lucky enough to be on the timer.

To be exact the bullet point reasons are:

- Non-intuitive to new players
- Bad profit per hour considering the 15-90 minute wait time and low amount of dead drop posters
- No additional incentive for NFSD to interact with the system other than selling off contraband crates
- Long boring wait time to get to the dead drops themselves
- Nobody except the smugglers can find the dead drop unless it's done on accident

## How to test


To test the systems this is what I did:

- Change the Content.Server/_NF/Smuggling/Components/DeadDropComponent.cs CoolDowns to their suggested value in the comment above them
- Start up the server
- Admin ghost to Tinnia's or other POIs
- Check any posters with DD appended to their internal name at the end and see whether NFSD and Freelance (1/3 chance) radio sparks up
- Spawn Urist McHands, headset with NFSD and Freelance encryption keys, Detective PDA or higher (Warden, Sheriff, SR, etc.) and forensic scanner (you need to be inside an entity for the DeadDropSystem to add the verb to the poster, this was unchanged behaviour)
- Afterwards get prints of the activated poster with a Detective ID or higher access to ensure it gives 3 FUC, make sure it can't give multiple rewards per same poster and that it only works for people with Detective access or higher (it's done this way so detectives don't lose their ability when they get promoted to "Captain" after buying a ship)
- Get to the dead drop location by going into B->Ghost, and then Admin->Objects->Syndicate Supply Drop->Right Click->Follow
- Spawn Urist with all his gear again there
- Scan any object inside the dead drop and ensure that it gives 6 FUC and doesn't give more than that if you scan more objects.
- Try to spawn more than 5 dead drops and see if there's more than 5, the oldest one should disappear (dequeue)


## Media



## Breaking changes


**Changelog**


:cl:
- tweak: More object types can be used as dead drops (more than posters).
- add: Hints for dead drop locations spawn randomly at round start and mid-round through faxes.
- add: The detective now has a module for their forensic scanner that gives spesos & FUCs on scanning dead drops and pods.
- add: Adds smuggling guidebook entry with a better (NFSD-side) explanation of mechanics.
- add: Both pirates and NFSD may receive the location of a drop pod about 15 minutes after a smuggling note is taken.
- add: Compromising a dead drop (by scanning or unanchoring it) moves it to another random location.
